crates/noirc_frontend/src/hir/mod.rs

```diff
@@ -66,19 +66,18 @@ impl Context {
     /// - Expects check_crate to be called beforehand
     /// - Panics if no main function is found
     pub fn get_main_function(&self, crate_id: &CrateId) -> Option<FuncId> {
-        // Find the local crate, one should always be present
-        let local_crate = self.def_map(crate_id).unwrap();
-
-        // Check the crate type
-        // We don't panic here to allow users to `evaluate` libraries which will do nothing
-        if self.crate_graph[*crate_id].crate_type == CrateType::Binary {
-            // All Binaries should have a main function
+        if let Some(local_crate) = self.def_map(crate_id) {
             local_crate.main_function()
         } else {
             None
         }
     }
 
+    pub fn is_binary_crate(&self, crate_id: &CrateId) -> bool {
+        // Currently, anything with a `main` function is a binary crate
+        matches!(self.get_main_function(crate_id), Some(_))
+    }
```

Comment:
A library (`lib.nr`) with a `main` function would be treated as a binary crate here.

Reply:
I think that is fine though; although, in rust, the extra `main` would be ignored and clippy would give a warning that it is unused. I could see us doing that, wdyt?

Reply:
This still has the concept of a crate-type in the frontend, its just that it is implicit by having a main function IIUC. I think its better to be explicit about the crate-type

Reply:
Clippy !== frontend in rust. Clippy is a linting tool that runs some phases of the compiler and then gives you feedback. The cargo abstraction defines what a lib or bin actually is, including specifying custom paths (you can specify `src/lib.rs` as the path for a `[[bin]]` iirc).